### PR TITLE
[15.0] [MIG] product_dimension

### PR DESCRIPTION
**Pull request by celm1990**

### CODE REVIEW

**Review by ajaniszewska-dev — APPROVED** [review]

File: product_dimension/models/product.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+            self.product_height,
+            self.product_width,
+            self.dimensional_uom_id,
+        )
+
+    @api.model
+    def _get_dimension_uom_domain(self):
+        return [("category_id", "=", self.env.ref("uom.uom_categ_length").id)]
+
+    product_length = fields.Float("length")
```
Comment (rousseldenis) [inline product_dimension/models/product.py]: Could you put fields definition on top ?

File: product_dimension/models/product.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    dimensional_uom_id = fields.Many2one(
+        "uom.uom",
+        "Dimensional UoM",
+        domain=lambda self: self._get_dimension_uom_domain(),
+        help="UoM for length, height, width",
+        default=lambda self: self.env.ref("uom.product_uom_meter"),
+    )
+
+
+class ProductTemplate(models.Model):
```
Comment (rousseldenis) [inline product_dimension/models/product.py]: Could you split one model per file ?

File: product_dimension/models/product.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+        return dimensional_uom._compute_quantity(
+            qty=measure,
+            to_unit=uom_meters,
+            round=False,
+        )
+
+    # Define all the related fields in product.template with 'readonly=False'
+    # to be able to modify the values from product.template.
+    dimensional_uom_id = fields.Many2one(
```
Comment (rousseldenis) [inline product_dimension/models/product.py]: Same as above

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Thanks for your contribution.

**Comment by i-vyshnevska** [comment]
pls consider to check discussion in this PR https://github.com/OCA/product-attribute/pull/957 before merging

**Review by simahawk — CHANGES_REQUESTED** [review]
waiting for https://github.com/OCA/product-attribute/pull/957 evolution

**Review by gurneyalex — APPROVED** [review]

**Comment by celm1990** [comment]
> waiting for #957 evolution

Hi @simahawk the other PR already is merge, but into this module I must be change field type from `Float `to `Integer`???
Is confused for me, because fields into `product.product` are Float(https://github.com/odoo/odoo/blob/15.0/addons/product/models/product.py#L124-L125), only on `stock.package.type `fields are Integer(https://github.com/odoo/odoo/blob/15.0/addons/stock/models/stock_package_type.py#L19-L20) but are models are independent

Thanks for your comments

**Comment by i-vyshnevska** [comment]
@celm1990 
https://github.com/OCA/product-attribute/pull/957 is about keeping the same type as in origin delivery https://github.com/odoo/odoo/blob/c9bfda40b48b3625765b7d220944938323350d0b/addons/delivery/models/product_packaging.py#L17
AFAIS this model was removed on v15 so I suppose you just need to rebase history and remove product_packaging.py file in the migration step

**Comment by celm1990** [comment]
> @celm1990
> #957 is about keeping the same type as in origin delivery https://github.com/odoo/odoo/blob/c9bfda40b48b3625765b7d220944938323350d0b/addons/delivery/models/product_packaging.py#L17
> AFAIS this model was removed on v15 so I suppose you just need to rebase history and remove product_packaging.py file in the migration step

`delivery `inherit `stock.packing.type`(https://github.com/odoo/odoo/blob/15.0/addons/stock/models/stock_package_type.py#L19-L20) instead of  `product.packaging`(https://github.com/odoo/odoo/blob/15.0/addons/product/models/product.py#L752) but both models exist on V15

Sorry but i not see because this module is related or need changes from other module

**Comment by i-vyshnevska** [comment]
@celm1990 sorry for misleading your PR about "product_dimension" https://github.com/OCA/product-attribute/pull/957 is about "product_packaging_dimension" addon, no direct relation here

**Review by i-vyshnevska — APPROVED** [review]

**Comment by celm1990** [comment]
@simahawk can you update your review please

**Review by simahawk — APPROVED** [review]
pls fix pre-commit

**Comment by celm1990** [comment]
> pls fix pre-commit

Sorry but I not understand what is error on pre-commit????
image

**Comment by celm1990** [comment]
ping @simahawk  @gurneyalex @rousseldenis

**Pull request merged**